### PR TITLE
Add option to define complete output path

### PR DESCRIPTION
**Pull request by harterSe**
Suggestion for #7

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -132,77 +130,90 @@ async function createRegularSprite(): Promise<void> {
...
   }
   // Compile the sprite
-  spriter.compile(function(err: Error, result: Array<PlainObjectType>): void {
+  spriter.compile((err: Error, result: Array<PlainObjectType>): void => {
     if (err) {
       throw err;
     }
     // @Todo(Chris): check if this is ever used
     // this has no effect as far as i can see
+    console.log(result)
```
Comment (christian-reichart) [inline src/index.ts]: remove console.log

@todo can be deleted aswell, leftover from typings merge i think

File: src/index.ts [hunk 2]
```diff
@@ -132,77 +130,90 @@ async function createRegularSprite(): Promise<void> {
...
+  const filePath = join(opts.dest, `${opts.name}.svg`)
+  const jsonSprite = getSvgJson(filePath) as INode;
+
   jsonSprite.children.forEach((symbol: INode) => {
-    modifyAttributes(symbol, new Map(), new Map());
+    modifyAttributes(symbol, opts);
   });
-  fs.writeFileSync(`${fullPath}${opts.subdirName}/${opts.fileName}.svg`, svgson.stringify(jsonSprite));
+
+  console.log(filePath);
```
Comment (christian-reichart) [inline src/index.ts]: remove console.log

File: README.md [hunk 3]
```diff
@@ -69,8 +69,8 @@ The creation of the chameleon sprite can be customized with various options. All
 ```javascript
 module.exports = {
   path: 'path/to/svg/directory/',           // default: '' (current working directory)
-  subdirName: 'my-sprite-dir',              // default: 'chameleon-sprite' (created inside your SVG directory, stores all generated files)
-  fileName: 'my-sprite',                    // default: 'chameleon-sprite' (used for .svg, .scss and .css files)
+  dest: 'path/to/dest/directory/'           // default: path + name as subfolder, if dest is specified no additional subfolder is created
+  name: 'my-sprite',                        // default: 'chameleon-sprite' (used for .svg, .scss and .css files)
   scss: true,                               // default: false (creates scss with classes for dimensions)
   css: true,                                // default: false (creates css with classes for dimensions)
```
Comment (christian-reichart) [inline README.md]: @harterSe dest on main object looks good, but the restructuring of the scss and css options is still todo! See my comment on the issue:

```
dimensionStyles: {
    css: {
      create: false, // automatically true if other option is set
      dest: null, // default same dir as sprite
      name: null // default is name
    },
    scss: {
      create: false, // automatically true if other option is set
      dest: null, // default same dir as sprite
      name: null // default is name
    },
  },
```
to keep the cli options simple i would leave --scss and --css to set dimensionStyles.scss/css.create  and expand it with --scss-dest, --scss-name, --css-dest, --css-name

**Review by christian-reichart — CHANGES_REQUESTED** [review]
hey @harterSe  thanks for taking this issue. Left you comments, we still need to do the scss and css dest stuff ;)

File: src/index.ts (hunk 1 above)
Comment (christian-reichart) [reply src/index.ts:173]: Ah, I see you left a todo for me, sorry. What do you mean, has no effect? The result contains information for svg-sprite on which modes were set active in the options (currently we only use symbol mode). https://www.npmjs.com/package/svg-sprite#usage-pattern

Comment (christian-reichart) [reply src/index.ts:173]: Ah I see, this comment isn't from you... just delete it with the console.log all together :D I think this was a leftover from the typings merge

File: src/index.ts [hunk 4]
```diff
@@ -311,7 +319,36 @@ function applyCustomOptions(customOptions: Partial<ChameleonOptions>): Chameleon
     customOptions.transition.apply = true;
   }
 
-  return deepMerge<ChameleonOptions>(getDefaultOptions(), customOptions);
+  console.log(customOptions.dimensionStyles)
```
Comment (christian-reichart) [inline src/index.ts]: remove console.log

File: src/index.ts [hunk 5]
```diff
@@ -311,7 +319,36 @@ function applyCustomOptions(customOptions: Partial<ChameleonOptions>): Chameleon
...
+
+  const scss = options.dimensionStyles.scss
+  const scssName = `${scss.name || options.name}.scss`;
+  const scssPath = getAbsolutePath(scss.dest || options.dest);
+
+  css.dest = join(cssPath, cssName);
+  scss.dest = join(scssPath, scssName);
+
+
+  console.log(options)
```
Comment (christian-reichart) [inline src/index.ts]: remove console.log

**Review by christian-reichart — CHANGES_REQUESTED** [review]
@harterSe  looks good to me! If you remove the two console logs i can merge this :)

File: src/index.ts (hunk 4 above)
Comment (harterSe) [reply src/index.ts]: Ok I will remove the logs this evening.

Comment (harterSe) [reply src/index.ts]: done

File: src/index.ts (hunk 5 above)
Comment (harterSe) [reply src/index.ts]: done

**Comment by christian-reichart** [comment]
@harterSe thanks for your contribution!

**Pull request merged**